lxd/storage/drivers/driver_dir.go

```diff
@@ -279,72 +278,61 @@ func (d *dir) CreateVolumeFromMigration(vol Volume, conn io.ReadWriteCloser, vol
 		}
 	}()
 
-	// Snapshots are sent first by the sender, so create these first.
-	for _, snapName := range volTargetArgs.Snapshots {
-		snapshot, err := vol.NewSnapshot(snapName)
-		if err != nil {
-			return err
-		}
-
-		snapPath := snapshot.MountPath()
-		err = snapshot.CreateMountPath()
-		if err != nil {
-			return err
-		}
+	// Create the main volume path.
+	volPath := vol.MountPath()
+	err = vol.CreateMountPath()
+	if err != nil {
+		return err
+	}
 
-		revertPaths = append(revertPaths, snapPath)
+	revertPaths = append(revertPaths, volPath)
 
-		// Initialise the snapshot's quota with the parent volume's ID.
-		err = d.initQuota(snapPath, volID)
-		if err != nil {
-			return err
-		}
+	// Ensure the volume is mounted.
+	err = vol.MountTask(func(mountPath string, op *operations.Operation) error {
+		path := shared.AddSlash(mountPath)
 
-		// Receive snapshot from sender (ensure local snapshot volume is mounted if needed).
-		err = snapshot.MountTask(func(mountPath string, op *operations.Operation) error {
+		// Snapshots are sent first by the sender, so create these first.
+		for _, snapName := range volTargetArgs.Snapshots {
+			// Receive the snapshot
 			var wrapper *ioprogress.ProgressTracker
 			if volTargetArgs.TrackProgress {
-				wrapper = migration.ProgressTracker(op, "fs_progress", snapshot.name)
+				wrapper = migration.ProgressTracker(op, "fs_progress", snapName)
```

Comment:
@stgraber the previous implementation created a snapshot volume struct from the parent using `snapshot, err := vol.NewSnapshot(snapName)` which allowed us to use the `snapshot.name` in the ProgressTracker, this would have included the parent name, e.g. "parent/snapshot". Would it be preferable to restore that or just have the snapshot name as it is now. I'm not familiar with what the progress tracker name is used for.